techempower/scripts/tfb-load.sh

```diff
@@ -0,0 +1,123 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+### Script to remove the galaxies setup ###
+#
+
+CURRENT_DIR="$(dirname "$(realpath "$0")")"
+source ${CURRENT_DIR}/tfb-common.sh
+
+function usage() {
+        echo
+        echo "Usage: -c CLUSTER_TYPE[minikube|openshift] [-i SERVER_INSTANCES] [-n NAMESPACE] [-a IP_ADDR]"
+        exit -1
+}
+
+SERVER_INSTANCES=1
+MAX_LOOP=4
+NAMESPACE=${DEFAULT_NAMESPACE}
+APP_NAME="tfb-qrh"
+END_POINT="db"
+THREADS=48
+DURATION=20
+
+while getopts c:i:a:n:-: gopts
+do
+        case ${gopts} in
+        -)
+                case "${OPTARG}" in
+                        iter=*)
+                                MAX_LOOP=${OPTARG#*=}
+                                ;;
+                esac
+                ;;
+        c)
+                CLUSTER_TYPE=${OPTARG}
+                ;;
+        i)
+                SERVER_INSTANCES="${OPTARG}"
+                ;;
+        a)
+                IP_ADDR="${OPTARG}"
+                ;;
+        n)
+                NAMESPACE="${OPTARG}"
+                ;;
+        esac
+done
+
+if [ -z "${CLUSTER_TYPE}" ]; then
+        usage
+fi
+
+case ${CLUSTER_TYPE} in
+minikube)
+        if [ -z "${IP_ADDR}" ]; then
+                IP_ADDR=$(minikube ip)
+        fi
+	TECHEMPOWER_PORT=$(kubectl -n ${NAMESPACE} get svc ${APP_NAME} --no-headers -o=custom-columns=PORT:.spec.ports[*].nodePort)
+        ;;
+openshift)
+        if [ -z "${IP_ADDR}" ]; then
+                IP_ADDR=($(oc status --namespace=${NAMESPACE} | grep ${APP_NAME} | grep port | cut -d " " -f1 | cut -d "/" -f3))
+        fi
+        ;;
+*)
+        echo "Load is not determined"
+        ;;
+esac
+
+
+LOG_DIR="${PWD}/logs/${APP_NAME}-$(date +%Y%m%d%H%M)"
+mkdir -p ${LOG_DIR}
+
+for(( inst=0; inst<${SERVER_INSTANCES}; inst++ ))
+do
+	# Check if the application is running
+        check_app
+	
+	SERVER_INSTANCE_PORT=${TECHEMPOWER_PORT[${inst}]}
+        for iter in `seq 1 ${MAX_LOOP}`
+        do
+		# Change these appropriately to vary load
+                CONNECTION=$(( 128*iter ))
+
+                echo
+                echo "#########################################################################################"
+                echo "                             Running with Connections ${connection}                      "
+                echo "#########################################################################################"
+                echo
+
+               # Run the wrk load
+                echo "Running wrk load for ${APP_NAME} instance ${inst} with the following parameters"
+		if [ "${CLUSTER_TYPE}" == "openshift" ]; then
+			cmd="${HYPERFOIL_DIR}/wrk.sh --latency --threads=${THREADS} --connections=${CONNECTION} --duration=${DURATION}s http://${IP_ADDR}/${END_POINT}"
```

Comment:
HYPERFOIL_DIR is not specified in the usage. Please document how to install hyperfoil in the readme.

Reply:
I see load_setup in the common script so I think calling it here is missing.